src/corehost/cli/fx_reference.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal.h"
+#include "fx_ver.h"
+#include "fx_reference.h"
+#include "roll_fwd_on_no_candidate_fx_option.h"
+
+bool fx_reference_t::is_forward_compatible(const fx_ver_t& other) const
+{
+    // We expect the version to be <
+    assert(get_fx_version_number() < other);
+
+    // Support >= cases for release
```

Comment:
The next line
```if (get_fx_version_number() >= other)```
could be changed to
```if (get_fx_version_number() == other)``` and just let the assert catch internal bugs.

Does this address your concern?

Other options are to change the assert to a throw, or return an error code instead of a bool. However I think those are overkill here.

Reply:
I like the assert only approach. Let it catch internal bugs only.